File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# Time:  O(n)
# Space: O(n)

from typing import List


# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookup = {}
        for i in range(len(nums)):
            if nums[i] in lookup:
                return lookup[nums[i]], i
            lookup[target - nums[i]] = i


# @lc code=end

File: 1.two-sum.js
/*
 * @lc app=leetcode id=1 lang=javascript
 *
 * [1] Two Sum
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const lookup = {};
  for (let i = 0; i < nums.length; ++i) {
    if (nums[i] in lookup) {
      return [lookup[nums[i]], i];
    }
    lookup[target - nums[i]] = i;
  }
};
// @lc code=end
